### PR TITLE
Add list component to SB - HTML & Angular

### LINKED ISSUE #2266: Add the List component to Angular storybook
**User Story:**
As Spark we want to add the List component to Angular storybook

**Notes:**
This was missed first time around.

**AC:**
- will see List component in Angular storybook

### LINKED ISSUE #2268: Add the List component to HTML storybook
**User Story:**
As Spark we want to add the List component to HTML storybook

**Notes:**
This was missed first time around.

**AC:**
- will see List component in HTML storybook

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
Adds the List component stories for HTML & Angular

### Associated Issue 
Fixes #2268 
Fixes #2266 

### Documentation
 - [x] Update Spark Docs Angular
 - [x] Update Spark Docs Vanilla

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Screenshots
Add screenshots to help explain your PR if you'd like. However, this is not
expected.

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import { SprkUnorderedListModule } from '../sprk-unordered-list/sprk-unordered-list.module';
+import { SprkListItemModule } from '../sprk-list-item/sprk-list-item.module';
+
+export default {
+  title: 'Components|List',
+  component: SprkUnorderedListModule,
+  decorators: [
+    storyWrapper(
+      storyContent => (
+        `<div class="sprk-o-Box sprk-u-mlm">${ storyContent }<div>`
```
Comment (RVMendoza) [inline angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts]: do we need this sprk-u-mlm?

Comment (guern1kn) [reply angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts]: Yes, the unordered list appears like this without it... 
Screen Shot 2019-10-31 at 1 27 39 PM

File: angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        Ordered List Item
+      </sprk-list-item>
+    </sprk-ordered-list>
+  `,
+});
+
+export const indented = () => ({
+  moduleMetadata: modules,
+  template: `
+    <p>
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts]: I know we had these paragraphs in there before, but they aren't related to the component itself. They are kinda confusing as to why they are there to people, of course we know we put them there for illustration purposes of the indentation, but that doesn't make sense in the SB.

File: angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts (hunk 1 above)
Comment (afebbraro) [reply angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts]: @guern1kn how come the angular SB needed the margin left and the HTML one didn't?

File: html/base/list.stories.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  `
+);
+
+export const indented = () => (
+  `
+    <p>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+      Sed gravida urna quis nulla ultrices, sed efficitur
+      risus elementum.
+    </p>
```
Comment (afebbraro) [inline html/base/list.stories.js]: Same thing about the paragraphs here with them not being part of the list component.

File: angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts (hunk 1 above)
Comment (guern1kn) [reply angular/projects/spark-angular/src/lib/components/sprk-ordered-list/sprk-list.stories.ts]: @ssabo1 Can we create a research issue to look into this? The angular component list items are rendering the text at the start of the padding, while the html is rendering the dot/number at the start of the padding. Not 100% sure if this is a bug or just how Angular works. 
image
image

@afebbraro

**Pull request merged**